### ISSUE TITLE
"Cannot find certificate" on Windows when EV signing

### ISSUE THREAD

**Issue by brandonhilkert** [electron-userland/electron-builder#4626]
* **Version**: 22.3.2

* **Target**: Win x64

We're upgrading from version `20.28.4`. Our previous build pipeline worked great. Just updated to `22.3.2` and we get the `Cannot find certificate` error now.

In `package.json`:
```
 "win": {
      "publisherName": "Bark Technologies, Inc",
      "certificateSubjectName": "Bark Technologies, Inc"
    }
```

```
PS C:\Users\bh> Get-ChildItem -Recurse Cert: -CodeSigningCert | Select-Object -Property Subject,PSParentPath,Thumbprint | ConvertTo-Json -Compress
{"Subject":"CN=\"Bark Technologies, Inc\", O=\"Bark Technologies, Inc\", L=Atlanta, S=Georgia, C=US, SERIALNUMBER=5783034, OID.2.5.4.15=Private Organization, OID.1.3.6.1.4.1.311.60.2.1.2=Delaware, OID.1.3.6.1.4.1.311.60.2.1.3=US","PSParentPath":"Microsoft.PowerShell.Security\\Certificate::CurrentUser\\My","Thumbprint":"E458FBF56BA4C15003C647EAE58B1D2FF8AA38E2"}
```

Any ideas?

**Comment by brandonhilkert** [comment]
Confirmed that `21.2.0` works as expected. So it seems the jump to `22` caused things to not work for us.

**Comment by Feverqwe** [comment]
I have same bug. Wrote about it here:
https://github.com/electron-userland/electron-builder/pull/4499#issuecomment-579254203

Now if set certificateSubjectName and certificateSha1 it will work, but it bug.

**Comment by kzimny** [comment]
I have exactly the same issue with v.22.3.1 and v.22.3.2. My EV code sign certificate contains umlaut character in company address and that seems cause the problem. The workaround is:
```
        "win": {
            "certificateSha1": "E458FBF56BA4C15003C647EAE58B1D2FF8AA38E2"
        },
```

**Comment by davej** [comment]
Also had this problem, my certificate did not contain any unusual characters. The `certificateSha1` (above) workaround solved it.

**Comment by tnishimura** [comment]
Setting `certificateSha1` worked for me as well. I did not have any unusual chars either (besides a comma...). Note that the hash needs to be ALL UPPER CASE. The Gemalto certificate manager for my Safenet 5110 key calls the hash the "thumbprint".

**Comment by mixmix** [comment]
I also used the `certificateSha1` to fix this, but it's totally unclear from the docs where that hash comes from?
I tried hashing my csc.pfx` certificate, that wasn't it?

Then reading the errors it said _something like_ 
```
Could not find matching certificate. Certificates we checked:
[
  { name, ...., certificateSha1 },
  { name, ...., certificateSha1 }    // << the one I want
]
```

so yeah I had a list of certificates and just grabbed the `certificateSha1` for the one that looked like a match in the Error. It felt pretty bad but it worked? :cry:
